src/apme_engine/graph/rules/L111_inventory_group_hyphens_graph.py

```
@@ -0,0 +1,370 @@
+    seen: set[str] = set()
+
+    try:
+        data = yaml.safe_load(content)
+        line_map = _get_yaml_line_map(content)
+        for group_name, line in _parse_yaml_groups(data, line_map=line_map):
+            if "-" in group_name and group_name not in seen:
+                seen.add(group_name)
+                groups_with_hyphens.append((group_name, line))
+    except yaml.YAMLError as err:
+        logger.debug("Failed to parse YAML inventory: %s", err)
+    except Exception as err:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)